src/data/nat/basic.lean

```diff
@@ -222,6 +222,18 @@ theorem le_zero_iff {i : ℕ} : i ≤ 0 ↔ i = 0 :=
 lemma zero_max {m : ℕ} : max 0 m = m :=
 max_eq_right (zero_le _)
 
+lemma add_eq_max_iff {n m : ℕ} :
+  n + m = max n m ↔ n = 0 ∨ m = 0 :=
+begin
+  split,
+  { intro h,
+    cases le_total n m with H H,
+    { simpa [H] using or.inl h },
+    { simpa [H] using or.inr h } },
+  { rintro (rfl|rfl);
+    simp }
+end
+
```

Comment:
A similar lemma (with the same proof script!) is:
```suggestion

lemma nat.min_eq_zero_iff {n m : ℕ} :
  min n m = 0 ↔ n = 0 ∨ m = 0 :=
begin
  split,
  { intro h,
    cases le_total n m with H H,
    { simpa [H] using or.inl h },
    { simpa [H] using or.inr h } },
  { rintro (rfl|rfl);
    simp }
end
```

Which can then be used to golf your other proof.